### PR TITLE
Milestone three

### PR DESCRIPTION
**Pull request by IanJeannin**

### CODE REVIEW

File: Assets/Scripts/DetectLookedAtInteractive.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    [Tooltip("How far from the raycast origin we will search for interactive elements")]
+    [SerializeField]
+    private float maxRange = 5.0f;
+
+    /// <summary>
+    /// Event raised when the player looks at a different IInteractive.
+    /// </summary>
+    public static event Action<IInteractive> LookedAtInteractiveChanged;
+
+    public IInteractive LookedAtInteractive
```
Comment (DirtyDanChicago) [inline Assets/Scripts/DetectLookedAtInteractive.cs:28]: I would comment this out as well, help describe what it does.

File: Assets/Scripts/Door.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
```
Comment (DirtyDanChicago) [inline Assets/Scripts/Door.cs:4]: Summary of the script would be nice.

File: Assets/Scripts/InteractiveObjects.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
```
Comment (DirtyDanChicago) [inline Assets/Scripts/InteractiveObjects.cs:4]: Summarize script.

File: Assets/Scripts/InteractiveObjects.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+[RequireComponent(typeof(AudioSource))]
+public class InteractiveObjects : MonoBehaviour, IInteractive
+{
+    [SerializeField]
+    protected string displayText = nameof(InteractiveObjects);
+
+    public string DisplayText => displayText;
+    private AudioSource audioSource;
+
+    protected virtual void Awake()
```
Comment (DirtyDanChicago) [inline Assets/Scripts/InteractiveObjects.cs:20]: I would comment on each method.

File: Assets/Scripts/LookedAtInteractiveDisplayText.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    private IInteractive lookedAtInteractive;
+    private Text displayText;
+
+    private void Awake()
+    {
+        displayText = GetComponent<Text>();
+        UpdateDisplayText();
+    }
+
+    private void UpdateDisplayText()
```
Comment (DirtyDanChicago) [inline Assets/Scripts/LookedAtInteractiveDisplayText.cs:27]: I'd start commenting each individual method.

File: Assets/Scripts/TitleMenu.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
+﻿using UnityEngine;
+using UnityEngine.SceneManagement;
+
```
Comment (DirtyDanChicago) [inline Assets/Scripts/TitleMenu.cs:3]: Summary of the script would be nice.

File: Assets/Scripts/ToggleSetActive.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ToggleSetActive : InteractiveObjects
```
Comment (DirtyDanChicago) [inline Assets/Scripts/ToggleSetActive.cs:8]: Summary of the script is nice, please and thank you.

**Review by DirtyDanChicago — CHANGES_REQUESTED** [review]
Really just make the changes I requested and everything should be okie dokie artichokie.

File: Assets/Scripts/DetectLookedAtInteractive.cs (hunk 1 above)
Comment (IanJeannin) [reply Assets/Scripts/DetectLookedAtInteractive.cs:28]: Done.

File: Assets/Scripts/Door.cs (hunk 2 above)
Comment (IanJeannin) [reply Assets/Scripts/Door.cs:4]: Done and Done.

File: Assets/Scripts/InteractiveObjects.cs (hunk 3 above)
Comment (IanJeannin) [reply Assets/Scripts/InteractiveObjects.cs:4]: Mission Accomplished.

File: Assets/Scripts/InteractiveObjects.cs (hunk 4 above)
Comment (IanJeannin) [reply Assets/Scripts/InteractiveObjects.cs:20]: I have completed the objective.

File: Assets/Scripts/LookedAtInteractiveDisplayText.cs (hunk 5 above)
Comment (IanJeannin) [reply Assets/Scripts/LookedAtInteractiveDisplayText.cs:27]: Started and Finished.

File: Assets/Scripts/TitleMenu.cs (hunk 6 above)
Comment (IanJeannin) [reply Assets/Scripts/TitleMenu.cs:3]: Summary added.

File: Assets/Scripts/ToggleSetActive.cs (hunk 7 above)
Comment (IanJeannin) [reply Assets/Scripts/ToggleSetActive.cs:8]: Niceties were added, you are most welcome.

**Comment by DirtyDanChicago** [comment]
Changes made, good work!

**Pull request merged**